### PR TITLE
IJPS-49

### PR DESCRIPTION
**Pull request by Thomas-ForniCGI**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/IJPS-49

### Change description ###
Build new DELETE /sittingRecord/{sittingRecordId} endpoint

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/integrationTest/java/uk/gov/hmcts/reform/jps/controllers/SittingRecordControllerITest.java [hunk 1]
```diff
@@ -173,4 +201,83 @@ void shouldReturn400ResponseWhenEnumValuesIncorrect() throws Exception {
...
     }
 
+    @ParameterizedTest
+    @CsvSource(textBlock = """
+      # ROLE
+      jps-recorder
+        """)
+    @Sql(scripts = {DELETE_SITTING_RECORD_STATUS_HISTORY, ADD_SITTING_RECORD_STATUS_HISTORY})
+    @WithMockUser(authorities = {"jps-recorder"})
+    void shouldDeleteSittingRecordWhenSittingRecordPresentRecorder(String role) throws Exception {
```
Comment (SomeshRA) [inline src/integrationTest/java/uk/gov/hmcts/reform/jps/controllers/SittingRecordControllerITest.java:211]: Please move to SittingRecordDeleteControllerITest

File: src/integrationTest/java/uk/gov/hmcts/reform/jps/controllers/SittingRecordControllerITest.java [hunk 2]
```diff
@@ -173,4 +201,83 @@ void shouldReturn400ResponseWhenEnumValuesIncorrect() throws Exception {
...
+    }
+
+    @ParameterizedTest
+    @CsvSource(textBlock = """
+      # ROLE
+      jps-submitter
+        """)
+    @Sql(scripts = {DELETE_SITTING_RECORD_STATUS_HISTORY, ADD_SITTING_RECORD_STATUS_HISTORY})
+    @WithMockUser(authorities = {"jps-submitter"})
+    void shouldDeleteSittingRecordWhenSittingRecordPresentSubmitter(String role) throws Exception {
```
Comment (SomeshRA) [inline src/integrationTest/java/uk/gov/hmcts/reform/jps/controllers/SittingRecordControllerITest.java:229]: Please move to SittingRecordDeleteControllerITest

File: src/integrationTest/java/uk/gov/hmcts/reform/jps/controllers/SittingRecordControllerITest.java [hunk 3]
```diff
@@ -173,4 +201,83 @@ void shouldReturn400ResponseWhenEnumValuesIncorrect() throws Exception {
...
+    }
+
+    @ParameterizedTest
+    @CsvSource(textBlock = """
+      # ROLE
+      jps-admin
+        """)
+    @Sql(scripts = {DELETE_SITTING_RECORD_STATUS_HISTORY, ADD_SITTING_RECORD_STATUS_HISTORY})
+    @WithMockUser(authorities = {"jps-admin"})
+    void shouldDeleteSittingRecordWhenSittingRecordPresentAdmin(String role) throws Exception {
```
Comment (SomeshRA) [inline src/integrationTest/java/uk/gov/hmcts/reform/jps/controllers/SittingRecordControllerITest.java:247]: Please move to SittingRecordDeleteControllerITest

File: src/integrationTest/java/uk/gov/hmcts/reform/jps/controllers/SittingRecordControllerITest.java [hunk 4]
```diff
@@ -173,4 +201,83 @@ void shouldReturn400ResponseWhenEnumValuesIncorrect() throws Exception {
...
+        when(userInfo.getName()).thenReturn("Joe Bloggs");
+
+        mockMvc.perform(MockMvcRequestBuilders.delete("/sittingRecord/{sittingRecordId}", 4))
+            .andDo(print())
+            .andExpect(status().isOk());
+    }
+
+    @Test
+    @WithMockUser(authorities = {"jps-recorder"})
+    void shouldThrowSittingRecordMandatoryWhenSittingRecordMissing() throws Exception {
```
Comment (SomeshRA) [inline src/integrationTest/java/uk/gov/hmcts/reform/jps/controllers/SittingRecordControllerITest.java:260]: Please move to SittingRecordDeleteControllerITest

File: src/integrationTest/java/uk/gov/hmcts/reform/jps/controllers/SittingRecordControllerITest.java [hunk 5]
```diff
@@ -173,4 +201,83 @@ void shouldReturn400ResponseWhenEnumValuesIncorrect() throws Exception {
...
+                status().isBadRequest(),
+                jsonPath("$.errors[0].fieldName").value("PathVariable"),
+                jsonPath("$.errors[0].message").value("sittingRecordId is mandatory")
+            );
+    }
+
+
+    @Test
+    @WithMockUser(authorities = {"jps-recorder"})
+    void shouldThrowSittingRecordNotFoundWhenSittingRecordNotFoundInDb() throws Exception {
```
Comment (SomeshRA) [inline src/integrationTest/java/uk/gov/hmcts/reform/jps/controllers/SittingRecordControllerITest.java:273]: Please move to SittingRecordDeleteControllerITest

File: src/integrationTest/java/uk/gov/hmcts/reform/jps/controllers/SittingRecordControllerITest.java [hunk 6]
```diff
@@ -173,4 +201,83 @@ void shouldReturn400ResponseWhenEnumValuesIncorrect() throws Exception {
             .contains("one of the values accepted for Enum class: [ASCENDING, DESCENDING]");
     }
 
+    @ParameterizedTest
```
Comment (SomeshRA) [inline src/integrationTest/java/uk/gov/hmcts/reform/jps/controllers/SittingRecordControllerITest.java:204]: ParameterizedTest can be removed

File: src/integrationTest/java/uk/gov/hmcts/reform/jps/controllers/SittingRecordControllerITest.java [hunk 7]
```diff
@@ -173,4 +201,83 @@ void shouldReturn400ResponseWhenEnumValuesIncorrect() throws Exception {
...
+        when(userInfo.getRoles()).thenReturn(List.of(role));
+        when(userInfo.getUid()).thenReturn("d139a314-eb40-45f4-9e7a-9e13f143cc3a");
+        when(userInfo.getName()).thenReturn("Joe Bloggs");
+
+        mockMvc.perform(MockMvcRequestBuilders.delete("/sittingRecord/{sittingRecordId}", 3))
+            .andDo(print())
+            .andExpect(status().isOk());
+    }
+
+    @ParameterizedTest
```
Comment (SomeshRA) [inline src/integrationTest/java/uk/gov/hmcts/reform/jps/controllers/SittingRecordControllerITest.java:222]: ParameterizedTest can be removed

File: src/integrationTest/java/uk/gov/hmcts/reform/jps/controllers/SittingRecordControllerITest.java [hunk 8]
```diff
@@ -173,4 +201,83 @@ void shouldReturn400ResponseWhenEnumValuesIncorrect() throws Exception {
...
+        when(userInfo.getRoles()).thenReturn(List.of(role));
+        when(userInfo.getUid()).thenReturn("d139a314-eb40-45f4-9e7a-9e13f143cc3a");
+        when(userInfo.getName()).thenReturn("Joe Bloggs");
+
+        mockMvc.perform(MockMvcRequestBuilders.delete("/sittingRecord/{sittingRecordId}", 2))
+            .andDo(print())
+            .andExpect(status().isOk());
+    }
+
+    @ParameterizedTest
```
Comment (SomeshRA) [inline src/integrationTest/java/uk/gov/hmcts/reform/jps/controllers/SittingRecordControllerITest.java:240]: ParameterizedTest can be removed

File: src/main/java/uk/gov/hmcts/reform/jps/data/SecurityUtils.java [hunk 9]
```diff
@@ -60,6 +61,10 @@ public String getUserId() {
         return getUserInfo().getUid();
     }
 
+    public List<String> getUserRoles() {
```
Comment (SomeshRA) [inline src/main/java/uk/gov/hmcts/reform/jps/data/SecurityUtils.java:64]: I suppose this method is not use

File: src/main/java/uk/gov/hmcts/reform/jps/services/SittingRecordService.java [hunk 10]
```diff
@@ -14,16 +20,45 @@
...
-    private final SittingRecordRepository sittingRecordRepository;
+
+    @Autowired
+    private SittingRecordRepository sittingRecordRepository;
+
+    @Autowired
+    private SecurityUtils securityUtils;
+
+
+    private Consumer<uk.gov.hmcts.reform.jps.domain.SittingRecord>
```
Comment (SomeshRA) [inline src/main/java/uk/gov/hmcts/reform/jps/services/SittingRecordService.java:44]: This is not in use

File: src/test/java/uk/gov/hmcts/reform/jps/controllers/SittingRecordControllerTest.java [hunk 11]
```diff
@@ -42,7 +46,7 @@
 import static org.testcontainers.shaded.com.google.common.base.Charsets.UTF_8;
 import static org.testcontainers.shaded.com.google.common.io.Resources.getResource;
 
-@WebMvcTest(controllers = SittingRecordController.class,
+@WebMvcTest(controllers = {SittingRecordController.class, SittingRecordDeleteController.class},
```
Comment (SomeshRA) [inline src/test/java/uk/gov/hmcts/reform/jps/controllers/SittingRecordControllerTest.java:49]: Create new SittingRecordDeleteControllerTest

File: src/test/java/uk/gov/hmcts/reform/jps/controllers/SittingRecordControllerTest.java [hunk 12]
```diff
@@ -216,4 +220,41 @@ void shouldReturnResponseWithSittingRecordsCountButNoSittingRecordsForRequestedO
         //verify(caseWorkerService).setCaseWorkerDetails(eq(sittingRecords));
 
     }
+
+    @Test
+    @WithMockUser(authorities = {"jps-recorder"})
```
Comment (SomeshRA) [inline src/test/java/uk/gov/hmcts/reform/jps/controllers/SittingRecordControllerTest.java:225]: Move to SittingRecordDeleteControllerTest

File: src/test/java/uk/gov/hmcts/reform/jps/controllers/SittingRecordControllerTest.java [hunk 13]
```diff
@@ -216,4 +220,41 @@ void shouldReturnResponseWithSittingRecordsCountButNoSittingRecordsForRequestedO
...
+    void shouldDeleteSittingRecordWhenSittingRecordPresent() throws Exception {
+        mockMvc.perform(MockMvcRequestBuilders.delete("/sittingRecord/{sittingRecordId}", 2))
+            .andDo(print())
+            .andExpect(status().isOk());
+    }
+
+    @Test
+    @WithMockUser(authorities = {"jps-recorder"})
+    void shouldThrowSittingRecordMandatoryWhenSittingRecordMissing() throws Exception {
```
Comment (SomeshRA) [inline src/test/java/uk/gov/hmcts/reform/jps/controllers/SittingRecordControllerTest.java:234]: Move to SittingRecordDeleteControllerTest

File: src/test/java/uk/gov/hmcts/reform/jps/controllers/SittingRecordControllerTest.java [hunk 14]
```diff
@@ -216,4 +220,41 @@ void shouldReturnResponseWithSittingRecordsCountButNoSittingRecordsForRequestedO
...
+                status().isBadRequest(),
+                jsonPath("$.errors[0].fieldName").value("PathVariable"),
+                jsonPath("$.errors[0].message").value("sittingRecordId is mandatory")
+            );
+    }
+
+
+    @Test
+    @WithMockUser(authorities = {"jps-recorder"})
+    void shouldThrowSittingRecordNotFoundWhenSittingRecordNotFoundInDb() throws Exception {
```
Comment (SomeshRA) [inline src/test/java/uk/gov/hmcts/reform/jps/controllers/SittingRecordControllerTest.java:247]: Move to SittingRecordDeleteControllerTest

File: src/test/java/uk/gov/hmcts/reform/jps/services/SittingRecordServiceTest.java [hunk 15]
```diff
@@ -190,30 +221,199 @@ void shouldSaveSittingRecordsWhenRequestIsValid() throws IOException {
...
+
+        Optional<StatusHistory> optionalStatusHistory
+            = sittingRecord.getStatusHistories().stream().max(Comparator.comparing(
+            StatusHistory::getChangeDateTime));
+        StatusHistory statusHistory = null;
+        if (optionalStatusHistory != null && !optionalStatusHistory.isEmpty()) {
+            statusHistory = optionalStatusHistory.get();
+        }
+
+        assertThat(statusHistory.getStatusId().equals("DELETED"));
```
Comment (SomeshRA) [inline src/test/java/uk/gov/hmcts/reform/jps/services/SittingRecordServiceTest.java:308]: Complete assertion assertThat(statusHistory.getStatusId()).isEqualTo("DELETED");

File: src/test/java/uk/gov/hmcts/reform/jps/services/SittingRecordServiceTest.java [hunk 16]
```diff
@@ -190,30 +221,199 @@ void shouldSaveSittingRecordsWhenRequestIsValid() throws IOException {
...
+            RECORDED.name()
+        );
+
+        Optional<StatusHistory> optionalStatusHistory = sittingRecord.getLatestStatusHistory();
+        StatusHistory statusHistory = null;
+        if (optionalStatusHistory != null && !optionalStatusHistory.isEmpty()) {
+            statusHistory = optionalStatusHistory.get();
+        }
+
+        assertThat(statusHistory.getStatusId().equals("DELETED"));
```
Comment (SomeshRA) [inline src/test/java/uk/gov/hmcts/reform/jps/services/SittingRecordServiceTest.java:327]: Complete assertion assertThat(statusHistory.getStatusId()).isEqualTo("DELETED");

File: src/test/java/uk/gov/hmcts/reform/jps/services/SittingRecordServiceTest.java [hunk 17]
```diff
@@ -190,30 +221,199 @@ void shouldSaveSittingRecordsWhenRequestIsValid() throws IOException {
...
+            SUBMITTED.name()
+        );
+
+        Optional<StatusHistory> optionalStatusHistory = sittingRecord.getLatestStatusHistory();
+        StatusHistory statusHistory = null;
+        if (optionalStatusHistory != null && !optionalStatusHistory.isEmpty()) {
+            statusHistory = optionalStatusHistory.get();
+        }
+
+        assertThat(statusHistory.getStatusId().equals("DELETED"));
```
Comment (SomeshRA) [inline src/test/java/uk/gov/hmcts/reform/jps/services/SittingRecordServiceTest.java:346]: Complete assertion assertThat(statusHistory.getStatusId()).isEqualTo("DELETED");

File: src/test/java/uk/gov/hmcts/reform/jps/services/SittingRecordServiceTest.java [hunk 18]
```diff
@@ -190,30 +221,199 @@ void shouldSaveSittingRecordsWhenRequestIsValid() throws IOException {
...
+            sittingRecordService.deleteSittingRecord(sittingRecord.getId());
+        });
+
+        Optional<StatusHistory> optionalStatusHistory = sittingRecord.getLatestStatusHistory();
+        StatusHistory statusHistory = null;
+        if (optionalStatusHistory != null && !optionalStatusHistory.isEmpty()) {
+            statusHistory = optionalStatusHistory.get();
+        }
+
+        assertThat(statusHistory.getStatusId().equals("RECORDED"));
```
Comment (SomeshRA) [inline src/test/java/uk/gov/hmcts/reform/jps/services/SittingRecordServiceTest.java:370]: Complete assertion assertThat(statusHistory.getStatusId()).isEqualTo("RECORDED");

File: src/main/java/uk/gov/hmcts/reform/jps/services/SittingRecordService.java [hunk 19]
```diff
@@ -14,16 +20,45 @@
...
 
 
 @RequiredArgsConstructor(onConstructor = @__(@Autowired))
 @Service
+@Slf4j
 public class SittingRecordService {
-    private final SittingRecordRepository sittingRecordRepository;
+
+    @Autowired
+    private SittingRecordRepository sittingRecordRepository;
```
Comment (SomeshRA) [inline src/main/java/uk/gov/hmcts/reform/jps/services/SittingRecordService.java:38]: We can either have autowire or required agrs constuctor

File: src/main/java/uk/gov/hmcts/reform/jps/controllers/SittingRecordDeleteController.java [hunk 20]
```diff
@@ -0,0 +1,41 @@
...
+@RequestMapping(
+    path = "/sittingRecord",
+    produces = MediaType.APPLICATION_JSON_VALUE
+)
+@RequiredArgsConstructor(onConstructor = @__(@Autowired))
+@Slf4j
+public class SittingRecordDeleteController {
+    private final SittingRecordService sittingRecordService;
+
+    @DeleteMapping(
```
Comment (SomeshRA) [inline src/main/java/uk/gov/hmcts/reform/jps/controllers/SittingRecordDeleteController.java:31]: Swagger info missing

**Review by SomeshRA — APPROVED** [review]
LGTM

**Review by RebeccaBaker — APPROVED** [review]

**Pull request merged**